### PR TITLE
Prevent crash in audio_backend/alsa.rs when switching from Kodi audio…

### PR DESCRIPTION
**Pull request by thekr1s**
… to librespot
This is just to prevent the crash. Causes start play to keep silent. After another pause and play of the user, the music starts..

### CODE REVIEW

**Comment by ComlOnline** [comment]
Sorry I lost my linux machine due to a hard drive failure on friday. Just testing this now.

Looks good though. Thanks

**Comment by thekr1s** [comment]
Error handling could be done better. There is currently no error reported to the user. I wonder if hat is possible, did not have a look at it...

File: src/audio_backend/alsa.rs [hunk 1]
```diff
@@ -18,10 +18,17 @@ impl Sink for AlsaSink {
     fn start(&mut self) -> io::Result<()> {
         if self.0.is_some() {
```
Comment (plietar) [inline src/audio_backend/alsa.rs]: This is obviously not your change, but can you make this an `is_none` and get rid of the else block

File: src/audio_backend/alsa.rs [hunk 2]
```diff
@@ -18,10 +18,17 @@ impl Sink for AlsaSink {
...
         } else {
-            self.0 = Some(PCM::open(&*self.1,
+            match PCM::open(&*self.1,
                                     Stream::Playback, Mode::Blocking,
                                     Format::Signed16, Access::Interleaved,
-                                    2, 44100).ok().unwrap());
+                                    2, 44100) {
+                Ok(f) => self.0 = Some(f),
+                Err(e) => {
+                    self.0 = None;
```
Comment (plietar) [inline src/audio_backend/alsa.rs]: `self.0` is already `None` here

File: src/audio_backend/alsa.rs (hunk 1 above)
Comment (thekr1s) [reply src/audio_backend/alsa.rs]: okee, I'll do

**Comment by thekr1s** [comment]
@sashahilton00 I should be more clear about the behavior.  Currently librespot crashes when open of the audio device fails. This open is done when pressing play on the spotify on the remote, my phone in this case.  My fix prevents a crash, but evenso the play command fails. In that case, the remote thinks librespot is playing the file.  So the the phone shows the song is playing, but it remains silent. Pressing pause and play again on the remote resolves the problem, and the song starts playing.

I would like to let librespot notify the uses on the remote that the play failed, but could not find an easy way to do that.  Therefore I would like the fix to be merged like it is now.

**Comment by sashahilton00** [comment]
@thekr1s how about sending a pause request, immediately followed by a play request, from within librespot when the error is encountered? A little hacky I know, but it should be fast enough that the user doesn't even see it, with the result that people don't think that the silent music without a crash is a bug, rather than intended behaviour?

**Comment by thekr1s** [comment]
Retrying to open the device immediately after an open failed is usesless, the secont open will also fail. Apparently it takes some time before Kodi closes the device. It appears that some mechanism in spotify is causing a retry of the play command every five seconds. So in the end it resolves automatically.  
No change on current pull request is required.

**Comment by sashahilton00** [comment]
Fair enough, merging.

**Pull request merged**